### ISSUE TITLE
Back end all menu items filter issue

### ISSUE THREAD

**Issue by jamield** [joomla/joomla-cms#19652]
### Steps to reproduce the issue

In administrator, go to menu -> All Menu Items. Click on a menu item and edit it in some way (I changed the title). Click save and close.

### Expected result

I would expect to go back to the list of all menu items (i.e. not filtered) so I can carry on going through all menu items.

### Actual result

I'm taken to the list of menu items, but it has been filtered by the menu that the item I was editing happens to be assigned to.

### System information (as much as possible)

Joomla 3.8.5. PHP 5.6. Ubuntu 14.04.1

### Additional comments

I understand that this is desired behaviour in certain situations. However, if I have chosen to view all menu items and edit one item, I wouldn't expect the system to decide on my behalf to then filter by the menu the item I just edited happens to sit in. It became quite frustrating when I was trying to go through all menu items making various changes and Joomla just kept deciding to filter the list!

**Comment by tonypartridge** [comment]
I agree what you are expecting is valid. If you want to filter that menu block the you select it, it shouldn't be assumed.

**Comment by Quy** [comment]
An intermediary solution is to open the link in a new tab. Edit it. Close the tab which will bring you back to previous tab with all the menu items.

**Comment by ghost** [comment]
Can this Issue be closed?

**Comment by jamield** [comment]
Has it been fixed?

**Comment by ReLater** [comment]
It won't be fixed in Joomla 3 because it's not a bug. It's the way Joomla does it; even if it's not always the "expected behavior" of a user. Some like it others not ;-)
Maybe something to be discussed for Joomla 4 ?

**Comment by ReLater** [comment]
Simply change the label ????

**Comment by jamield** [comment]
To be totally honest it's responses like this that drive people to
WordPress and other CMS's that take usability seriously.

What you're effectively saying is "yeah sorry it doesn't do what you would
logically expect, but tough. It's not a bug. Put up with it".

Close it if you want.

**Comment by brianteeman** [comment]
It's a bug.

**Comment by tonypartridge** [comment]
I’ll see if I can do a PR tomorrow. It is a bug, should be solvable by
adding in a query parameter for editing or similar.

**Comment by ReLater** [comment]
Why is this a bug? Because YOU say that? If you create a new menu item and return to the overview you'll see it filtered by the menu you've selected in the new menu item. Is that a bug, too, then? I hope that behavior (while creating new items) won't be changed.

**Comment by jamield** [comment]
No that wouldn't be a bug, and that's not the behaviour that's I'm
suggesting should be changed. And it shouldn't.

From your response, I'm not sure you've understood the issue here?

**Comment by ReLater** [comment]
I understood well. I just want to go sure that the filtering mentioned in https://github.com/joomla/joomla-cms/issues/19652#issuecomment-500151315 won't be changed because we're talking about the same saving routine in both cases here. "If you save go back and filter the overview by the selected menu item of the saved item."

And this behavior was not established erroneously but willfully. It's an "old feature" and since there's an agreement that features won't be changed in Joomla 3 (even if lots of them are discussible) because porting them to Joomla 4 would become to labor-intensive... 

If @tonypartridge finds a way, all fine...

**Comment by tonypartridge** [comment]
If you are viewing ALL menu items and edit one then save & close you should
be returned to the view that you were on which would be ALL menu items and
Not returned back to a filtered list of menu items from the menu of the
current menu item you are editing is on. Thats just completely illogical.

**Comment by brianteeman** [comment]
Can someone change the label to J4 please as its still reproducable

**Comment by coolcat-creations** [comment]
It's still the same in J5
